backoffice/backoffice/workflows/airflow_utils.py

```diff
@@ -158,9 +158,14 @@ def restart_workflow_dags(workflow_id, workflow_type, params=None):
     """
     executed_dags_for_workflow = find_executed_dags(workflow_id, workflow_type)
 
-    for dag_id in executed_dags_for_workflow:
+    data = None
+
+    for dag_id, dag_data in executed_dags_for_workflow.items():
         delete_workflow_dag(dag_id, str(workflow_id))
+        # fetch data from previous drag run
+        if data is None:
+            data = dag_data["conf"].get("data")
```

Comment:
Even with this fix, if params is not `None`, `data` will never be passed. I would suggest refactoring it a bit like the following

```suggestion

    for dag_id, dag_data in executed_dags_for_workflow.items():
        delete_workflow_dag(dag_id, str(workflow_id))
        dag_data = dag_data["conf"].get("data")
        if dag_data:
          return trigger_airflow(......, dag_data)
       return trigger_airflow(......, params)
```

Reply:
Okay. My reasoning is that if params are not `None` it means we want to completely ignore the `data` we were running the dag with before,  and use this new params as the new `data`. Is this not the case?

Also in the suggestion if we run an `AUTHOR_CREATE` workflow it will run both the `initialize` dag, and the `approve/reject` dag is this want we want or should the decision be completely reverted?